src/dodal/devices/pressure_jump_cell.py

```diff
@@ -213,19 +236,22 @@ def __init__(
         self.pump = Pump(f"{prefix}{cell_prefix}", name)
 
         self.pressure_transducer_1 = PressureTransducer(
-            prefix + "PP1:",
+            prefix=f"{prefix}{cell_prefix}",
             name="Pressure Transducer 1",
             adc_prefix=f"{prefix}{adc_prefix}-01:",
+            number=1
         )
         self.pressure_transducer_2 = PressureTransducer(
-            prefix + "PP2:",
+            prefix=f"{prefix}{cell_prefix}",
             name="Pressure Transducer 2",
             adc_prefix=f"{prefix}{adc_prefix}-02:",
+            number=2
         )
         self.pressure_transducer_3 = PressureTransducer(
-            prefix + "PP3:",
+            prefix=f"{prefix}{cell_prefix}",
             name="Pressure Transducer 3",
             adc_prefix=f"{prefix}{adc_prefix}-01:",
+            number=3
```

Comment:
might: there might be some broader dodal convention on how to handle the prefixes so that the nesting is predictable.

I am not sure
 @DominicOram  what would you do here?

Reply:
So I would do:
```python
class PressureJumpCell(StandardReadable):
    """
    High pressure X-ray cell, used to apply pressure or pressure jumps to a sample.
    """

    def __init__(
        self,
        prefix: str = "",
        cell_prefix: str = "",
        adc_prefix: str = "",
        name: str = "",
    ):
        self.all_valves_control = AllValvesControl(f"{prefix}{cell_prefix}", name)
        self.pump = Pump(f"{prefix}{cell_prefix}", name)

        self.pressure_transducers: DeviceVector[PressureTransducer] = DeviceVector(
            {
                i: PressureTransducer(
                    prefix=f"{prefix}{cell_prefix}",
                    number=i,
                    adc_prefix=f"{prefix}{adc_prefix}-{i:02}:",
                )
                for i in range(1, 4)
            }
        )
```
You can omit the name as the `DeviceVector` should fill it in with a sensible name after the fact.

Reply:
ah of course